physioqc/metrics/utils.py

```diff
@@ -85,3 +91,349 @@ def has_peaks(signal: Physio) -> bool:
+            Fs,
+            upperpass,
+            len(inputdata),
+            order,
+        )
+    sos = signal.butter(order, 2.0 * upperpass, btype="lowpass", output="sos", fs=Fs)
+    return signal.sosfiltfilt(sos, inputdata).real
+
+
+# @conditionaljit()
+def butterhpfiltfilt(Fs, lowerpass, inputdata, order, debug=False):
+    r"""Performs a bidirectional (zero phase) Butterworth highpass filter on an input vector
+    and returns the result.  Ends are padded to reduce transients.
+
+    Parameters
+    ----------
+    Fs : float
+        Sample rate in Hz
+        :param Fs:
+
+    lowerpass : float
+        Lower end of passband in Hz
+        :param lowerpass:
+
+    inputdata : 1D numpy array
+        Input data to be filtered
+        :param inputdata:
+
+    order : int
+        Order of Butterworth filter.
+        :param order:
+
+    debug : boolean, optional
+        When True, internal states of the function will be printed to help debugging.
+        :param debug:
+
+    Returns
+    -------
+    filtereddata : 1D float array
+        The filtered data
+    """
+    if lowerpass < 0.0:
+        lowerpass = 0.0
+    if debug:
+        print(
+            "butterhpfiltfilt - Fs, lowerpass, len(inputdata), order:",
+            Fs,
+            lowerpass,
+            len(inputdata),
+            order,
+        )
+    sos = signal.butter(order, 2.0 * lowerpass, btype="highpass", output="sos", fs=Fs)
+    return signal.sosfiltfilt(sos, inputdata).real
+
+
+# @conditionaljit()
+def butterbpfiltfilt(Fs, lowerpass, upperpass, inputdata, order, debug=False):
+    r"""Performs a bidirectional (zero phase) Butterworth bandpass filter on an input vector
+    and returns the result.  Ends are padded to reduce transients.
+
+    Parameters
+    ----------
+    Fs : float
+        Sample rate in Hz
+        :param Fs:
+
+    lowerpass : float
+        Lower end of passband in Hz
+        :param lowerpass:
+
+    upperpass : float
+        Upper end of passband in Hz
+        :param upperpass:
+
+    inputdata : 1D numpy array
+        Input data to be filtered
+        :param inputdata:
+
+    order : int
+        Order of Butterworth filter.
+        :param order:
+
+    debug : boolean, optional
+        When True, internal states of the function will be printed to help debugging.
+        :param debug:
+
+    Returns
+    -------
+    filtereddata : 1D float array
+        The filtered data
+    """
+    if upperpass > Fs / 2.0:
+        upperpass = Fs / 2.0
+    if lowerpass < 0.0:
+        lowerpass = 0.0
+    if debug:
+        print(
+            f"butterbpfiltfilt - {Fs=}, {lowerpass=}, {upperpass=}, {len(inputdata)=}, {order=}"
+        )
+    sos = signal.butter(
+        order, [2.0 * lowerpass, 2.0 * upperpass], btype="bandpass", output="sos", fs=Fs
+    )
+    if debug:
+        print(sos)
+    return signal.sosfiltfilt(sos, inputdata).real
+
+
+def readbidstsv(inputfilename, colspec=None, warn=True, debug=False):
```

Comment:
Really happy to see physioqc supporting BIDS dataset !!! :smile_cat: 

I'm wondering if there is a reason not to add [pybids](https://bids-standard.github.io/pybids/api.html) as a dependency, which might facilitate the loading data from a bids dataset part

Reply:
I can check that out - I just pulled in a routine I wrote a while back to use in one of my packages to read BIDS files, but I can use a standard routine.